<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>651</width>
    <height>840</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="btnFile">
           <property name="text">
            <string>测试用文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editFilename"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>字节序</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QRadioButton" name="radio_LittleEndian">
              <property name="text">
               <string>LittleEndian</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radio_BigEndian">
              <property name="text">
               <string>BigEndian</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>浮点数精度</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QRadioButton" name="radio_Single">
              <property name="text">
               <string>SinglePrecision</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radio_Double">
              <property name="text">
               <string>DoublePrecision</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>预编码数据序列化方式读写测试</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>qint8(-128~127)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_int8">
         <property name="minimum">
          <number>-128</number>
         </property>
         <property name="maximum">
          <number>127</number>
         </property>
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QPushButton" name="btnInt8_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QPushButton" name="btnInt8_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="0" column="6">
        <widget class="QLineEdit" name="editInt8"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>quint8(0~255)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_uint8">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>236</number>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="btnUint8_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QPushButton" name="btnUint8_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="1" column="6">
        <widget class="QLineEdit" name="editUint8"/>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>qint16(-32768~32767)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_int16">
         <property name="minimum">
          <number>-32768</number>
         </property>
         <property name="maximum">
          <number>32767</number>
         </property>
         <property name="value">
          <number>-1522</number>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QPushButton" name="btnInt16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5">
        <widget class="QPushButton" name="btnInt16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="2" column="6">
        <widget class="QLineEdit" name="editInt16"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>quint16(0~65535)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_uint16">
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>54862</number>
         </property>
        </widget>
       </item>
       <item row="3" column="4">
        <widget class="QPushButton" name="btnUint16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="3" column="5">
        <widget class="QPushButton" name="btnUint16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="3" column="6">
        <widget class="QLineEdit" name="editUint16"/>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>qint32</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_int32">
         <property name="minimum">
          <number>-65536</number>
         </property>
         <property name="maximum">
          <number>65536</number>
         </property>
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QPushButton" name="btnInt32_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="4" column="5">
        <widget class="QPushButton" name="btnInt32_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="4" column="6">
        <widget class="QLineEdit" name="editInt32"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>qint64</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="5" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_int64">
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="5" column="4">
        <widget class="QPushButton" name="btnInt64_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="5" column="5">
        <widget class="QPushButton" name="btnInt64_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="5" column="6">
        <widget class="QLineEdit" name="editInt64"/>
       </item>
       <item row="6" column="2">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Bool</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="6" column="4">
        <widget class="QPushButton" name="btnBool_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="5">
        <widget class="QPushButton" name="btnBool_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>qfloat16</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_float16">
         <property name="value">
          <number>99</number>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QPushButton" name="btnfloat16_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="7" column="5">
        <widget class="QPushButton" name="btnfloat16_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="7" column="6">
        <widget class="QLineEdit" name="editFloat16"/>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>float</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="8" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_float">
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QPushButton" name="btnFloat_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="8" column="5">
        <widget class="QPushButton" name="btnFloat_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="8" column="6">
        <widget class="QLineEdit" name="editFloat"/>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>double</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="9" column="2" colspan="2">
        <widget class="QSpinBox" name="spinBox_double">
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="9" column="4">
        <widget class="QPushButton" name="btnDouble_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="9" column="5">
        <widget class="QPushButton" name="btnDouble_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="9" column="6">
        <widget class="QLineEdit" name="editDouble"/>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>char*  string</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="10" column="1" colspan="3">
        <widget class="QLineEdit" name="editString_In">
         <property name="text">
          <string>Hello</string>
         </property>
        </widget>
       </item>
       <item row="10" column="4">
        <widget class="QPushButton" name="btnString_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="10" column="5">
        <widget class="QPushButton" name="btnString_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="10" column="6">
        <widget class="QLineEdit" name="editString"/>
       </item>
       <item row="11" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>QString</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="11" column="1" colspan="3">
        <widget class="QLineEdit" name="editQString_In">
         <property name="text">
          <string>Hello</string>
         </property>
        </widget>
       </item>
       <item row="11" column="4">
        <widget class="QPushButton" name="btnQString_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="11" column="5">
        <widget class="QPushButton" name="btnQString_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="11" column="6">
        <widget class="QLineEdit" name="editQString"/>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>QFont</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="1" colspan="3">
        <widget class="QPushButton" name="btnFont_In">
         <property name="text">
          <string>选择字体</string>
         </property>
        </widget>
       </item>
       <item row="12" column="4">
        <widget class="QPushButton" name="btnQFont_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="12" column="5">
        <widget class="QPushButton" name="btnQFont_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="12" column="6">
        <widget class="QLineEdit" name="editFont">
         <property name="text">
          <string>字体</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>QColor</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="13" column="1" colspan="3">
        <widget class="QPushButton" name="btnColor_In">
         <property name="text">
          <string>选择颜色</string>
         </property>
        </widget>
       </item>
       <item row="13" column="4">
        <widget class="QPushButton" name="btnQColor_Write">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="13" column="5">
        <widget class="QPushButton" name="btnQColor_Read">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="13" column="6">
        <widget class="QLineEdit" name="editColor">
         <property name="text">
          <string>颜色</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QCheckBox" name="chkBox_In">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="6" column="6">
        <widget class="QCheckBox" name="chkBox_Out">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actClearOutPut"/>
   <addaction name="actSaveAll"/>
   <addaction name="actReadAll"/>
   <addaction name="separator"/>
   <addaction name="actClose"/>
  </widget>
  <action name="actClearOutPut">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon/images/clearFile.png</normaloff>:/icon/images/clearFile.png</iconset>
   </property>
   <property name="text">
    <string>编辑框全清除</string>
   </property>
   <property name="toolTip">
    <string>读出数据全清</string>
   </property>
  </action>
  <action name="actClose">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon/images/exit.png</normaloff>:/icon/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
  </action>
  <action name="actSaveAll">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon/images/copy.png</normaloff>:/icon/images/copy.png</iconset>
   </property>
   <property name="text">
    <string>连续写入文件</string>
   </property>
   <property name="toolTip">
    <string>连续写入文件</string>
   </property>
  </action>
  <action name="actReadAll">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon/images/file-file.png</normaloff>:/icon/images/file-file.png</iconset>
   </property>
   <property name="text">
    <string>连续从文件读取</string>
   </property>
   <property name="toolTip">
    <string>连续从文件读取</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
